packages/tinygo/build.sh

```diff
@@ -0,0 +1,148 @@
+
+	make build/release \
+		LLVM_BUILDDIR="${TERMUX_PKG_HOSTBUILD_DIR}" \
+		CLANG="${TERMUX_PKG_HOSTBUILD_DIR}/bin/clang" \
+		LLVM_AR="${TERMUX_PKG_HOSTBUILD_DIR}/bin/llvm-ar" \
+		LLVM_NM="${TERMUX_PKG_HOSTBUILD_DIR}/bin/llvm-nm" \
+		USE_SYSTEM_BINARYEN=1
+	popd
+
+	local PATCHELF_BUILD_SH="${TERMUX_SCRIPTDIR}/packages/patchelf/build.sh"
+	local PATCHELF_SRCURL=$(. "${PATCHELF_BUILD_SH}"; echo $TERMUX_PKG_SRCURL)
+	local PATCHELF_SHA256=$(. "${PATCHELF_BUILD_SH}"; echo $TERMUX_PKG_SHA256)
+	local PATCHELF_TARFILE="${TERMUX_PKG_CACHEDIR}/$(basename "${PATCHELF_SRCURL}")"
+	termux_download "${PATCHELF_SRCURL}" "${PATCHELF_TARFILE}" "${PATCHELF_SHA256}"
+	local PATCHELF_SRCDIR="${TERMUX_PKG_HOSTBUILD_DIR}/_patchelf"
+	mkdir -p "${PATCHELF_SRCDIR}"
+	tar xf "${PATCHELF_TARFILE}" -C "${PATCHELF_SRCDIR}" --strip-components=1
+	pushd "${PATCHELF_SRCDIR}"
+	./bootstrap.sh
+	./configure
+	make -j "${TERMUX_MAKE_PROCESSES}"
+	popd
+}
+
+termux_step_make() {
+	termux_setup_golang
+	termux_setup_cmake
+	termux_setup_ninja
+
+	# from packages/libllvm/build.sh
+	export _LLVM_DEFAULT_TARGET_TRIPLE=${CCTERMUX_HOST_PLATFORM/-/-unknown-}
+	export _LLVM_TARGET_ARCH
+	if [[ "${TERMUX_ARCH}" == "arm" ]]; then
+		_LLVM_TARGET_ARCH=ARM
+	elif [[ "${TERMUX_ARCH}" == "aarch64" ]]; then
+		_LLVM_TARGET_ARCH=AArch64
+	elif [[ "${TERMUX_ARCH}" == "i686" ]] || [[ "${TERMUX_ARCH}" == "x86_64" ]]; then
+		_LLVM_TARGET_ARCH=X86
+	else
+		termux_error_exit "Invalid arch: ${TERMUX_ARCH}"
+	fi
+
+	_LLVM_OPTION+=" -DLLVM_TARGET_ARCH=${_LLVM_TARGET_ARCH}"
+	_LLVM_OPTION+=" -DLLVM_HOST_TRIPLE=${_LLVM_DEFAULT_TARGET_TRIPLE}"
+
+	make llvm-build LLVM_OPTION="$(echo ${_LLVM_OPTION})"
+
+	ninja -C llvm-build \
+		-j"${TERMUX_MAKE_PROCESSES}" \
+		${_LLVM_EXTRA_BUILD_TARGETS}
+
+	# replace Android llvm-config with wrapper around host build
+	cat <<- EOF > llvm-build/bin/llvm-config
+	#!/bin/bash
+	${TERMUX_PKG_HOSTBUILD_DIR}/bin/llvm-config "\$@" | \
+	sed \
+	-e "s|${TERMUX_PKG_HOSTBUILD_DIR}|${TERMUX_PKG_SRCDIR}/llvm-build|g" \
+	-e "s|-lrt|-lc|g"
+	EOF
+
+	make tinygo
+	mkdir -p build/release/tinygo/bin
+	cp -fv build/tinygo build/release/tinygo/bin
+
+	# skip make gen-device, done in host build
+	# skip make wasi-libc, NDK doesnt support wasm32-unknown-wasi
+	# skip make binaryen
+
+	# TODO investigate later, fix excessive runpath entries
+	readelf -dW build/release/tinygo/bin/tinygo | grep RUNPATH
+	"${TERMUX_PKG_HOSTBUILD_DIR}/_patchelf/src/patchelf" \
+		--set-rpath "${TERMUX_PREFIX}/lib" \
+		build/release/tinygo/bin/tinygo
+	readelf -dW build/release/tinygo/bin/tinygo | grep RUNPATH
+}
+
+termux_step_make_install() {
+	mkdir -p "${TERMUX_PREFIX}/lib/tinygo"
+	cp -fr "${TERMUX_PKG_SRCDIR}/build/release/tinygo" "${TERMUX_PREFIX}/lib"
+	ln -sv "../lib/tinygo/bin/tinygo" "${TERMUX_PREFIX}/bin/tinygo"
+}
+
+termux_step_install_license() {
+	install -Dm644 -t "${TERMUX_PREFIX}/share/doc/${TERMUX_PKG_NAME}" \
+		"${TERMUX_PKG_SRCDIR}/LICENSE"
+}
```

Comment:
Please refrain from unnecessarily re-define `termux_step_install_license` function. It should be sufficient to define

```bash
TERMUX_PKG_LICENSE_FILE="LICENSE"
```